service/history/api/create_workflow_util.go

```diff
@@ -146,17 +161,13 @@ func GenerateFirstWorkflowTask(
 	mutableState workflow.MutableState,
 	parentInfo *workflowspb.ParentExecutionInfo,
 	startEvent *historypb.HistoryEvent,
-) error {
-
+	bypassTaskGeneration bool,
+) (int64, error) {
 	if parentInfo == nil {
 		// WorkflowTask is only created when it is not a Child Workflow and no backoff is needed
-		if err := mutableState.AddFirstWorkflowTaskScheduled(
-			startEvent,
-		); err != nil {
-			return err
-		}
+		return mutableState.AddFirstWorkflowTaskScheduled(startEvent, bypassTaskGeneration)
```

Comment:
can it do AddFirstWorkflowTaskScheduled(...) then GetNextEventID()

Reply:
Sure, but that's less convenient IMHO.

Reply:
This is how the Core SDK API for constructing histories looks like.
I vote to keep this change.

Reply:
i thought it could be easier to handle the 0 schedule event id. As long as we handle the 0 event id, it is fine.